text/0000-rust-governance.md

```diff
@@ -0,0 +1,728 @@
+which should be prioritized.
+
+### Moderation team
+
+Finally, the moderation team is responsible for dealing with CoC violations.
+
+One key difference from the other subteams is that the moderation team does not
+have a leader. Its members are chosen directly by the core team, and should be
+community members who have demonstrated the highest standard of discourse and
+maturity. To limit conflicts of interest, **the moderation subteam should not
+include any core team members**. However, the subteam is free to consult with
+the core team as it deems appropriate.
+
+The moderation team will have a public email address that can be used to raise
+complaints about CoC violations (forwards to all active moderators).
+
+#### Initial plan for moderation
+
+What follows is an initial proposal for the mechanics of moderation. The
+moderation subteam may choose to revise this proposal by drafting an RFC, which
+will be approved by the core team.
+
+Moderation begins whenever a moderator becomes aware of a CoC problem, either
+through a complaint or by observing it directly. In general, the enforcement
+steps are as follows:
+
+> **These steps are adapted from text written by Manish Goregaokar, who helped
+articulate them from experience as a Stack Exchange moderator.**
+
+* Except for extreme cases (see below), try first to address the problem with a
+  light public comment on thread, aimed to de-escalate the situation. These
+  comments should strive for as much empathy as possible. Moderators should
+  emphasize that dissenting opinions are valued, and strive to ensure that the
+  technical points are heard even as they work to cool things down.
+
+  When a discussion has just gotten a bit heated, the comment can just be a
+  reminder to be respectful and that there is rarely a clear "right" answer. In
+  cases that are more clearly over the line into personal attacks, it can
+  directly call out a problematic comment.
+
+* If the problem persists on thread, or if a particular person repeatedly comes
+  close to or steps over the line of a CoC violation, moderators then email the
+  offender privately. The message should include relevant portions of the CoC
+  together with the offending comments. Again, the goal is to de-escalate, and
+  the email should be written in a dispassionate and empathetic way. However,
+  the message should also make clear that continued violations may result in a
+  ban.
+
+* If problems still persist, the moderators can ban the offender. Banning should
+  occur for progressively longer periods, for example starting at 1 day, then 1
+  week, then permanent. The moderation subteam will determine the precise
+  guidelines here.
+
+In general, moderators can and should unilaterally take the first step, but
+steps beyond that (particularly banning) should be done via consensus with the
+other moderators. Permanent bans require core team approval.
```

Comment:
I did realise it was only permanent bans. I agree they ought to be rare and that is one reason I'm more keen to sort things out ahead of time, whereas for other things I think 'see how it goes' is a better approach. I worry that in the past the core team and core team + extras have been very hesitant around bans, when in retrospect they should have been handed out earlier. This extra check seems like it will encourage such hesitancy in the future too - more steps to go through and more people to debate means more chance of indecision. Hopefully, it will be rare enough to not matter....